c/extras/dt_ftrl.cc

```diff
@@ -341,23 +333,13 @@ double Ftrl::logloss(double p, bool y) {
   double epsilon = std::numeric_limits<double>::epsilon();
   p = std::max(std::min(p, 1 - epsilon), epsilon);
   if (y) {
-    return -log(p);
+    return -std::log(p);
   } else {
-    return -log(1 - p);
+    return -std::log(1 - p);
```

Comment:
Logloss can also be made branchless, though at the cost of several additional arithmetic operations:
 `- log(y * ( 2 * p - 1) - p + 1)`, not sure which one is faster. Since logloss is only needed for some kind of reporting, may be fully refactored later depending on requirements, we can probably leave it as is for simplicity.

Reply:
Ok. I changed it to `-log(p * (2*y - 1) - y + 1)` because `y` is integer, and probably it's easier for the compiler to do integer math